http-api/src/lib.rs

```diff
@@ -750,6 +807,25 @@ fn remote_branch(branch_name: &str, peer_id: &PeerId) -> git::Branch {
     )
 }
 
+fn commit_hash_to_fingerprint(ctx: Context, sha1: &str) -> Option<String> {
+    let output = Command::new("git")
+        .current_dir(ctx.paths.git_dir()) // We need to place the command execution in the git dir
+        .args(["show", sha1, "--pretty=%GF", "--raw"])
+        .output()
+        .expect("failed to execute git show command");
+    let result = output.stdout;
+
+    // If there is no stdout or if there is some stderr we return None early.
+    if result.is_empty() || !output.stderr.is_empty() {
+        return None;
+    }
+
+    BufReader::new(result.as_slice())
+        .lines()
+        .next()
+        .map(|line| line.unwrap_or_default())
```

Comment:
This should return an error, not an empty string, if the line cannot be parsed. I wouldn't use `map` here because that'll make it harder to exit on error.